### ISSUE TITLE
Question about training algorithm: why COCO?

### ISSUE THREAD

**Issue by ed353** [lengstrom/fast-style-transfer#186]
Hi Fast Style Transfer developers,

I like your code base because it's well-organized and easy to understand - thank you for writing it! I have a question about your training algorithm - why do you iterate over 82k images from the COCO dataset when training? From reading Gatys et. al., it seems like their loss function is only defined in terms of a single content image and a single style image. Why bother with the huge dataset? As I understand it, style transfer doesn't involve training (i.e. changing the weights of) a neural network but rather "training" an image to match a given style+content. 

If there's something I'm missing please let me know! I haven't read the other papers you cite; if I come across the answer to this question I'll post it and close the ticket.

Best,
Emily

**Comment by JEF1056** [comment]
Neural style is as you say, to train a neural network to find patterns between images and apply one to the other. However, the training can take long periods of time, and saved networks for one style image cannot be applied to multiple content images; the network simply cannot apply the pattern form the style to multiple content images, since it hasn't been trained on a wide enough set of content images.

So to answer your question, the COCO dataset images are different enough from one another, and provide enough variety in content that the network can apply its "style" to the new image quickly and actually produce good results.

Gatys' focuses on single image transfer, no matter how long it takes, but Fast-Style hopes to be able to apply styles quickly and reliably, almost in real-time.

**Comment by ed353** [comment]
OK, that explanation helps. Thank you Jess!
